salesforce/bulkwrite.go

```diff
@@ -0,0 +1,299 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return c.putCSV(ctx, location, file)
+}
+
+func (c *Connector) completeUpload(ctx context.Context, jobId string) (map[string]interface{}, error) {
+	updateLoadCompleteBody := map[string]interface{}{
+		"state": "UploadComplete",
+	}
+
+	location, err := joinURLPath(c.BaseURL, fmt.Sprintf("jobs/ingest/%s", jobId))
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.patch(ctx, location, updateLoadCompleteBody)
+	if err != nil {
+		return nil, fmt.Errorf("patch failed: %w", errors.Join(err, common.ErrRequestFailed))
+	}
+
+	return ParseAjsonNodeToMap(res)
+}
+
+func ParseAjsonNodeToMap(node *ajson.Node) (map[string]interface{}, error) {
+	parsed := map[string]interface{}{}
+
+	for _, key := range node.Keys() {
+		data, err := node.GetKey(key)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %s", ErrKeyNotFound, key)
+		}
+
+		switch data.Type() {
+		case ajson.Null:
+			parsed[key] = data.MustNull()
+		case ajson.Numeric:
+			parsed[key] = data.MustNumeric()
+		case ajson.String:
+			parsed[key] = data.MustString()
+		case ajson.Bool:
+			parsed[key] = data.MustBool()
+		case ajson.Array:
+			parsed[key] = data.MustArray()
+		case ajson.Object:
+			parsed[key] = data.MustObject()
+		default:
+			return nil, fmt.Errorf("%w: %d", ErrUnknownNodeType, data.Type())
+		}
+	}
+
+	return parsed, nil
+}
+
+type GetJobInfoResult struct {
+	Id                     string  `json:"id"`
+	Object                 string  `json:"object"`
+	CreatedById            string  `json:"createdById"`
+	ExternalIdFieldName    string  `json:"externalIdFieldName"`
+	State                  string  `json:"state"`
+	Operation              string  `json:"operation"`
+	ColumnDelimiter        string  `json:"columnDelimiter"`
+	LineEnding             string  `json:"lineEnding"`
+	NumberRecordsFailed    float64 `json:"numberRecordsFailed"`
+	NumberRecordsProcessed float64 `json:"numberRecordsProcessed"`
+
+	ApexProcessingTime      float64 `json:"apexProcessingTime"`
+	ApiActiveProcessingTime float64 `json:"apiActiveProcessingTime"`
+	ApiVersion              float64 `json:"apiVersion"`
+	ConcurrencyMode         string  `json:"concurrencyMode"`
+	ContentType             string  `json:"contentType"`
+	CreatedDate             string  `json:"createdDate"`
+	JobType                 string  `json:"jobType"`
+	Retries                 float64 `json:"retries"`
+	SystemModstamp          string  `json:"systemModstamp"`
+	TotalProcessingTime     float64 `json:"totalProcessingTime"`
+}
+
+func (c *Connector) GetJobInfo(ctx context.Context, jobId string) (*GetJobInfoResult, error) {
+	location, err := joinURLPath(c.BaseURL, "jobs/ingest", jobId)
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := c.get(ctx, location)
+	if err != nil {
+		return nil, fmt.Errorf("getGetInfo failed: %w", errors.Join(err, common.ErrRequestFailed))
+	}
+
+	dataMap, err := ParseAjsonNodeToMap(node)
```

Comment:
Since you've already defined a struct for GetJobInfoResult, this will be a lot less lines and won't require type assertion if you do the following:
- call https://pkg.go.dev/github.com/spyzhov/ajson#Marshal to turn node into a byte array
- call [json.Unmarshall](https://pkg.go.dev/encoding/json#Unmarshal) to turn the byte array into GetInfoJobResult